stl/src/dllmain.cpp

```diff
@@ -7,10 +7,11 @@
 
 #ifdef _CRT_APP
 // free static resource used by causality
-extern "C" void __cdecl __crtCleanupCausalityStaticFactories();
+extern "C" void __cdecl __crtCleanupCausalityStaticFactories() noexcept;
 #endif // defined(_CRT_APP)
 
-extern "C" BOOL APIENTRY DllMain(HMODULE /* hModule */, DWORD ul_reason_for_call, [[maybe_unused]] LPVOID lpReserved) {
+extern "C" BOOL APIENTRY DllMain(
+    HMODULE /* hModule */, DWORD ul_reason_for_call, [[maybe_unused]] LPVOID lpReserved) noexcept {

```

Comment:
#4087 is removing `__crtCleanupCausalityStaticFactories()` and changing `DllMain()` to be a stub. I'll pre-emptively resolve this conflict by dropping this PR's changes here, and adding `noexcept` to `DllMain()` in #4087.

Reply:
I got confused and mentioned this PR itself in the commit message instead of #4087, oopsy. The change itself is fine though.